src/client/webpack.config.js

```diff
@@ -0,0 +1,97 @@
+const { resolve: _resolve, sep } = require('path');
+const webpack = require('webpack');
+const HtmlWebpackRootPlugin = require('html-webpack-root-plugin');
+const TSConfigPathsPlugin = require('tsconfig-paths-webpack-plugin');
+const TerserPlugin = require('terser-webpack-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+
+const paths = {
+  src: _resolve(__dirname) + sep,
+  dist: {
+    rootDir: _resolve(__dirname, '..', '..', 'build'),
+    app: _resolve(__dirname, '..', '..', 'build', 'static'),
+  },
+};
+
+/**
+ * Client webpack build configuration.
+ *
+ * This webpack config produces a bundle for the client-side application only.
+ *
+ * @param {object} webpackEnv Webpack env object (basically any/all options passed in via the CLI)
+ * @param {object} [processEnv] Process env object (environment variables from process.env)
+ */
+const config = ({ mode = 'none' }, processEnv = {}) => ({
```

Comment:
So, this was meant to match the way that webpack expects you to export functional configs. According to [the documentation](https://v4.webpack.js.org/configuration/configuration-types/#exporting-a-function) the function recieves two arguments from webpack: `env` and `argv`. Passing the value down like that (rather than just using `process.env` directly) helps us to have a better flow of data so there's just one single source of config data.

In answering this question though you made me realize that actually the arguments have somehow gotten flipped, meaning that `mode` was _always_ defaulting to `'development'`. So I've fixed that in e37526ff8245cb027a0ba1ca0565125de28cf1cc

Reply:
So, this was meant to match the way that webpack expects you to export functional configs. According to [the documentation](https://v4.webpack.js.org/configuration/configuration-types/#exporting-a-function) the function recieves two arguments from webpack: `env` and `argv`. Passing the value down like that (rather than just using `process.env` directly) helps us to have a better flow of data so there's just one single source of config data.

In answering this question though you made me realize that actually the arguments have somehow gotten flipped, meaning that `mode` was _always_ defaulting to `'development'`. So I've fixed that in e37526ff8245cb027a0ba1ca0565125de28cf1cc